pontus/week2/Ass3/app.js

```diff
@@ -0,0 +1,241 @@
+class Wrapper extends React.Component { 
+	constructor(props) { 
+		super(props);
+
+      this.state = { lists:{} };
+
+		this.createList = () => { 
+			let newID = Date.now();
+			let lists = JSON.parse(JSON.stringify(this.state.lists));
+
+			if (this.refs.inputField.value != '') { 
+				let nameOfList = this.refs.inputField.value;
+				lists[newID] = { title: nameOfList };
+			}
+			else { 
+				lists[newID] = {};
+			}
+			this.setState({ lists });
+			this.refs.inputField.value = "";
+		}
+	}
+
+	moveMsg(currentListID, newListID, msgID) { 
+		let lists = JSON.parse(JSON.stringify(this.state.lists));
+
+		lists[newListID][msgID] = lists[currentListID][msgID];
+		delete lists[currentListID][msgID];
+		this.setState({ lists });
+	}
+
+	createMsg(listID, message) {
+		let newID = Date.now();
+		let msg = {text: message};
+		let lists = JSON.parse(JSON.stringify(this.state.lists));
+
+		lists[listID][newID] = msg;
+		this.setState({ lists });
+	}
+
+	deleteMsg(listID, msgID) { 
+		let lists = JSON.parse(JSON.stringify(this.state.lists));
+
+		delete lists[listID][msgID];
+		this.setState({ lists });
+	}
+
+	archiveMsg(listID, msgID) { 
+		let lists = JSON.parse(JSON.stringify(this.state.lists));
+
+		// toggle the isArchived state
+		lists[listID][msgID].isArchived = (!lists[listID][msgID].isArchived);
+		this.setState({ lists });
+	}
+
+	render() {
+		let lists = [];
+
+		for (let list in this.state.lists) { 
+			lists.push(<NamedList lists={this.state.lists} moveMsg={this.moveMsg.bind(this)} deleteMsg={this.deleteMsg.bind(this)} title={this.state.lists[list].title} archiveMsg={this.archiveMsg.bind(this)} createMsg={this.createMsg.bind(this)} listID={list} list={this.state.lists[list]} />);
```

Comment:
for the future, when having that many props then arrange your props per line to make it more readable:
```javascript
<NamedList
           lists={this.state.lists}
           moveMsg={this.moveMsg.bind(this)}
           deleteMsg={this.deleteMsg.bind(this)}
           title={this.state.lists[list].title}
           archiveMsg={this.archiveMsg.bind(this)}
           createMsg={this.createMsg.bind(this)}
           listID={list} list={this.state.lists[list]}
/>
```